app/config_test.go

```diff
@@ -4,67 +4,29 @@
 package app
 
 import (
-	"io/ioutil"
 	"strconv"
-	"strings"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 
 	"github.com/mattermost/mattermost-server/model"
 	"github.com/mattermost/mattermost-server/store/sqlstore"
 	"github.com/mattermost/mattermost-server/utils"
-	"github.com/mattermost/mattermost-server/utils/fileutils"
 )
 
-func TestLoadConfig(t *testing.T) {
-	tempConfig, err := ioutil.TempFile("", "")
-	require.Nil(t, err)
-
-	input, err := ioutil.ReadFile(fileutils.FindConfigFile("config.json"))
-	require.Nil(t, err)
-	lines := strings.Split(string(input), "\n")
-	for i, line := range lines {
-		if strings.Contains(line, "SiteURL") {
-			lines[i] = `        "SiteURL": "http://localhost:8065/",`
-		}
-	}
-	output := strings.Join(lines, "\n")
-	err = ioutil.WriteFile(tempConfig.Name(), []byte(output), 0644)
-	require.Nil(t, err)
-	tempConfig.Close()
-
-	a := App{
-		Srv: &Server{},
-	}
-	appErr := a.LoadConfig(tempConfig.Name())
-	require.Nil(t, appErr)
-
-	assert.Equal(t, "http://localhost:8065", *a.Config().ServiceSettings.SiteURL)
-}
-
 func TestConfigListener(t *testing.T) {
 	th := Setup().InitBasic()
 	defer th.TearDown()
 
 	originalSiteName := th.App.Config().TeamSettings.SiteName
-	th.App.UpdateConfig(func(cfg *model.Config) {
-		*cfg.TeamSettings.SiteName = "test123"
-	})
 
 	listenerCalled := false
 	listener := func(oldConfig *model.Config, newConfig *model.Config) {
-		if listenerCalled {
-			t.Fatal("listener called twice")
-		}
+		assert.False(t, listenerCalled, "listener called twice")
 
-		if *oldConfig.TeamSettings.SiteName != "test123" {
-			t.Fatal("old config contains incorrect site name")
-		} else if *newConfig.TeamSettings.SiteName != *originalSiteName {
-			t.Fatal("new config contains incorrect site name")
-		}
+		assert.Equal(t, *originalSiteName, *oldConfig.TeamSettings.SiteName, "old config contains incorrect site name")
```

Comment:
Observe that these assertions are exactly the opposite of the previous ones: the behaviour before relied on the ability to invoke `ReloadConfig()` and discard the assignment in `UpdateConfig` above, having not done a `PersistConfig`. I'm skeptical that this was the purpose of this test, given that we don't test listeners in a more "normal" way, so I opted to make this test more lifelike.

This was also the start of an attempt to eliminate the need to `PersistConfig` at all, but I ran into other difficulties that precluded making that change right now.